keypom/keypom-airfoil
# All drops and drop manager pagination, Custom drops type

# Description

- [x] Categorizes custom drops that are not created from the app as 'Other'
- [x] Added page query param to the url to retain pagination memory for All drops and Drop manager pages
- [x] Refactor drop managers (reduce duplicated codes and abstracted the common methods in DropManager component)

Approx 300 lines of redundant codes deleted 
![image](https://user-images.githubusercontent.com/19740800/227356439-b48707b2-0a96-437c-b08b-8f631f5639cf.png)


Fixes #191 and #190 

# How to test
Show steps to replicate and test the feature/fixes in this PR

1. Go to https://ericw-fix-all-drops-paginati.keypom-airfoil.pages.dev/drops
2. Paginate through all drops and ensure there are no missing drops
3. Click on a token/nft/ticket drop and ensure pagination and functionalities works as before
4. Refresh the page when page is paginated away from first page, ensure that the page displays the paginated keys/drops correctly
5. Find a custom drop and go on the drop manager page

# Screenshots/Screen Recording of Implementation
![image](https://user-images.githubusercontent.com/19740800/227341585-7e9124c5-05c7-4aab-b282-47eccce98a27.png)
![image](https://user-images.githubusercontent.com/19740800/227354632-e941f29f-4e1f-4743-a625-b003b2408f73.png)
![image](https://user-images.githubusercontent.com/19740800/227358211-27b319bc-7dc4-4d20-ab54-bda553684e79.png)
